src/barril/units/_scalar.py

```diff
@@ -313,8 +316,19 @@ def _DoOperation(self, p1, p2, operation, callback_operation):
             )
 
         unit_database = self._unit_database
-        operation = getattr(unit_database, operation)
-        q, v = operation(p1.GetQuantity(), p2.GetQuantity(), self._value, p2.value)
+        operation_func = getattr(unit_database, operation)
+        if p1_is_number:
+            assert operation in [
```

Comment:
Maybe raising an explicit `TypeError` here is better that an `assert`.

Reply:
True, I'll change it

Reply:
Hm after double checking the code, it is a sanity check, that condition should never be reached. It is mainly for ensure the code continues correct after some refatoring,